raiden/raiden_event_handler.py

```diff
@@ -202,7 +202,10 @@ def handle_paymentsentsuccess(
         # With the introduction of the lock we should always get
         # here only once per identifier so payment_status should always exist
         # see: https://github.com/raiden-network/raiden/pull/3191
-        payment_status.payment_done.set(True)
+        payment_status.payment_done.set({
+            'secret': payment_status.secret,
+            'secret_hash': payment_status.secret_hash,
```

Comment:
@offerm this comment is not related to your PR.

What looks wrong to me here is that `secret` and `secret_hash` have to be communicated through the result when the `payment_status` instance already has them. I looked around the code and it is mostly because we return the `AsyncResult` and not the `PaymentStatus` instance. This makes the addition of response items to the payload only be possible like what is done above. This could be improved by returning the payment_status instead of the result.

@LefterisJP, @hackaugusto thoughts?

Reply:
@rakanalh my 2 cents:
while not related directly to this PR, I initially tried to use the values from `payment_status` by taking it from `raiden.targets_to_identifiers_to_statuses[target]` after the AsyncResult is set. This failed since  the code in line 200 pops the payment_status from ` raiden.targets_to_identifiers_to_statuses[target]`.

What might be possible is to do `payment_status.payment_done.set(payment_status)` and have the full payload available after the `wait`.

Reply:
Yes, the pop could cause an issue. But i was commenting on the possibility to change here:
https://github.com/raiden-network/raiden/blob/63ace6a3c4ae62dd7d04e2ee776bb0ff1a6a8d76/raiden/api/python.py#L610-L617
Where the payment status could be returned and we only wait on the `payment_done` in this case. But the caller would still reference the payment status without having to look it up from the `targets_to_identifiers_to_statuses`.

Reply:
I agree. Thanks

Reply:
> Where the payment status could be returned and we only wait on the payment_done in this case

Where would we wait on that?

So essentially `python_api.transfer_and_wait()` would no longer exist with your suggestion?